File: tests/c/TestAbstractFactory.c
#define main pattern_main
#include "../../oop_in_c/AbstractFactory/AbstractFactory.c"
#undef main
#include <stdio.h>

int main() {
    printf("Testing C AbstractFactory...\n");
    // Call pattern main
    pattern_main();
    printf("C AbstractFactory Test Passed!\n");
    return 0;
}

File: tests/java/TestAbstractFactory.java
public class TestAbstractFactory {
  public static void main(String[] args) {
    System.out.println("Testing Java AbstractFactory...");
    try {
      Program.main(args);
    } catch (Exception e) {
      e.printStackTrace();
      System.exit(1);
    }
    System.out.println("Java AbstractFactory Test Passed!");
  }
}
